dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -0,0 +1,620 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+from segmentation_test_utils import make_batch_select_masks
+from PIL import Image
+from nose.tools import nottest
+import nvidia.dali.ops as ops
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import numpy as np
+import test_utils
+import inspect
+import os
+import math
+
+"""
+How to test variable (iter-to-iter) batch size for a given op?
+-------------------------------------------------------------------------------
+The idea is to create a Pipeline that assumes i2i variability, run 2 iterations
+and compare them with ad-hoc created Pipelines for given (constant) batch sizes.
+This can be easily done using `check_batch` function below.
+
+On top of that, there are some utility functions and routines to help with some
+common cases:
+1. If the operator is typically processing image-like data (i.e. 3-dim, uint8,
+   0-255, with shape like [640, 480, 3]) and you want to test default arguments
+   only, please add a record to the `ops_image_default_args` list
+2. If the operator is typically processing image-like data (i.e. 3-dim, uint8,
+   0-255, with shape like [640, 480, 3]) and you want to specify any number of
```

Comment:
How about more demanding scenarios (where maybe the shape is a bit more fuzzy?).
Also, didn't look for it yet, but do you test stuff like FCHW sequences? There might be some weirdness in sequence processing operators.

Reply:
Added sequence tests